### ISSUE TITLE
Authorize-Attribute producing white page / empty claims

### ISSUE THREAD

**Issue by Compufreak345** [chrisdrobison/aspnetcore-wsfed#1]
Hi,

first of all thanks for the great work - it's really appreciated and I am currently working on using your solution productively ;) 

I can finish the authorization process and get redirected to the ASP.NET-page after I succesfully logged in - but it seems the identity gets lost afterwards, probably because the SET-Cookie-Command that should appear after the login is not reaching my browser.  If I add an [Authorize]-Attribute I only get a 401-response, seeing a blank page. If I try to access the Claims-View without the Authorize-Attribute I don't get any claims, and in the AccountController/Login-Method User.Identities.Any(identity => identity.IsAuthenticated) always returns false - so I get redirected to the login-page of the ADFS, that immediately redirects me back to the application.

I tried to replicate what you did in your sample application in a blank .NET Core Web-Application - maybe I missed something.

Some snippets from my code : 
ConfigureServices: 
```
services.AddAuthentication(
                options => options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme);
```

Configure:
```
 var adfsSettings = Configuration.GetSection("ADFS");
            app.UseCookieAuthentication(new CookieAuthenticationOptions());
            app.UseWsFederationAuthentication(new WsFederationAuthenticationOptions
            {
                Wtrealm = adfsSettings.GetValue<string>("Wtrealm"),
                MetadataAddress =
                   adfsSettings.GetValue<string>("Metadata")
            });
```

I copied your Views & Controllers into my project.

I also need to do some ssl-redirection, but I don't think it is related to it : 
```
var sslPort = 0;
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseBrowserLink();
                // https://long2know.com/2016/07/asp-net-core-enforcing-https/
                var builder = new ConfigurationBuilder()
                        .SetBasePath(env.ContentRootPath)
                        .AddJsonFile(@"Properties/launchSettings.json", optional: false, reloadOnChange: true);
                var launchConfig = builder.Build();
                sslPort = launchConfig.GetValue<int>("iisSettings:iisExpress:sslPort");
            }
            else
            {
                app.UseExceptionHandler("/Home/Error");
            }

// https://long2know.com/2016/07/asp-net-core-enforcing-https/
            app.Use(async (context, next) =>
            {
                if (context.Request.IsHttps)
                {
                    await next();
                }
                else
                {
                    var sslPortStr = sslPort == 0 || sslPort == 443 ? string.Empty : $":{sslPort}";
                    var httpsUrl = $"https://{context.Request.Host.Host}{sslPortStr}{context.Request.Path}";
                    context.Response.Redirect(httpsUrl);
                }
            });
```

This are my dependencies: 
![image](https://cloud.githubusercontent.com/assets/10026694/24197937/fcc68584-0f03-11e7-8f44-088ad06e30e0.png)

Any ideas what I might be missing?

Thanks & best regards,
Christoph

**Comment by chrisdrobison** [comment]
Looks good to me. The only thing I'm not doing is that SSL redirect. Can
you capture the traffic in the Chrome dev tools (enable Preserve Logs) or
something similar so I can see what's getting passed back and forth?

**Comment by chrisdrobison** [comment]
The important call to look for is the post back to /signin-wsfed. That should be where the claims are read in and handed off to the middleware that stores them--in this case the cookie middleware.

**Comment by Compufreak345** [comment]
Read the part about the important call just yet, after doing my professional screenshots xD So, I think screenshots 7,8 & 9 might be the most interesting ones.

**Comment by chrisdrobison** [comment]
Numbers 7 & 8 doesn't look like the POST is going to /signin-wsfed, which is where the post would need to go for those claims to get picked up. On the WsFederationOptions object, you can try setting CallbackPath to something else. I'm guessing something is out of sync between an ADFS setting and the defaults this middleware puts in place. What does the list of reply urls look like in ADFS? Also, on number 1, is there a wreply attribute? In the end, ADFS does need to POST back to whatever the configured CallbackPath is so that the claims can get picked up.

**Comment by chrisdrobison** [comment]
It may be that ADFS is POSTing to the wtrealm (I can't be sure though, it has been a long time since I've used on-prem ADFS) in the absence of a wreply attribute, which is something Azure AD does not do in my sample. I'll have to look when I get on my computer a little later.

**Comment by Compufreak345** [comment]
Unfortunately we are currently not administrating the ADFS I am using, so I don't know what the attributes you ask for are currently set to. We are working on setting up our own Test-ADFS.
But, fortunately I found a workaround for now : 
```
app.UseWsFederationAuthentication(new WsFederationAuthenticationOptions
            {
                Wtrealm = adfsSettings.GetValue<string>("Wtrealm"),
                MetadataAddress =
                   adfsSettings.GetValue<string>("Metadata"),
                Wreply = adfsSettings.GetValue<string>("Wtrealm") + "/signin-wsfed"
            });
```

Just as a sidenote: When I set up the ADFS with the .NET Framework 4.6.2 Web-App-Wizard it automagically worked without me setting this parameter. So it seems to do something similar in the background. Maybe setting the default Wreply to match the default CallbackPath might be a solution. But I'm by far no ADFS-specialist to know what is the correct solution ;)

I'm really thankful to have a working ADFS in .NET Core now, this is awesome!
